Global Sidebar - update color scheme and show icons

Addison-Stavlo opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to # https://github.com/Automattic/dotcom-forge/issues/6143, https://github.com/Automattic/dotcom-forge/issues/6144, https://github.com/Automattic/dotcom-forge/issues/6145, https://github.com/Automattic/dotcom-forge/issues/6146

## Proposed Changes

* update the color scheme per designs linked in issue
* add previously used icons back in place.
* migrates sidebar styles to a consistent use of `--color-sidebar-*` variables, so switching color schemes in the future should be a bit easier. Going back to our previous dark mode now should only requires changing a few variable definitions. (do I hear a dark mode toggle coming the future? 👂 😆  )

<img width="481" alt="Screenshot 2024-03-22 at 1 44 08 PM" src="https://github.com/Automattic/wp-calypso/assets/28742426/e0dbefba-a57b-4f23-b98d-02534cbdcdd1">


## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Load the calypso live link and test the global nav around reader, /me, and /sites.  Global site view is also a concern functionality wise, but this is only seen by early access users and is due to disappear with the new tab approach.
* Sidebar should continue to work as expected.
* Verify icons are in place.
* Test hover, focus, selected styles, as well as for submenu groups like in the reader.

BONUS
- this is not a requirement/goal of this PR, but we are in a much better state to change colors in the future (or adapt toggles / color schemes).
- Test changing the color variables under the global sidebar selectors at the top of client/mysites/sidebar/style.scss .. to change the background, text colors, hover colors, icon colors etc.
- You should be able to make a custom color scheme, or swap back to dark mode scheme, simply by only changing these variables.
- known limitations at this point - we don't have a specific color for hover state of icons - when selected they take on the selected text color, when hovered they use the unselected text color. Also, the WordPress.com link in the header is currently set as a background image - if we instead used the SVG we could have that take different colors from css as well. Those seem like the last things that we would need to fix if we wanted to supply a more broad selection of color schemes to users.

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

Addison-Stavlo committed:
init new styles attempt

Addison-Stavlo committed:
add sidebar icons back in

Addison-Stavlo committed:
update styles for icon colors

Addison-Stavlo committed:
add icons back to global view

Addison-Stavlo committed:
Merge branch 'trunk' into update/global-nav-background-color

Addison-Stavlo committed:
update submenu background

Addison-Stavlo committed:
fix reader icon clip path

Addison-Stavlo committed:
add react component support for custom icon wrapper and use core WP icon for menu item

Addison-Stavlo committed:
update to use sidebar variables, and apply subtle border

Addison-Stavlo committed:
more color fixes

davemart-in commented:
A few notes:

1) Hover color here feel a tad strong:

![CleanShot 2024-03-22 at 09 15 55@2x](https://github.com/Automattic/wp-calypso/assets/5634774/e79ad003-f5a6-412f-822f-f4d204301032)

![CleanShot 2024-03-22 at 09 18 51@2x](https://github.com/Automattic/wp-calypso/assets/5634774/d3c34bad-b773-4c6b-8df1-2e2ebc434f8b)

Feels like we may want a new color in-between 0 and 5. In my prototype, I had `--studio-gray-2-5: #eee;` which feels like a good compromise.

2) These my profile icons should probably be dark, not white on hover:

![CleanShot 2024-03-22 at 09 19 36@2x](https://github.com/Automattic/wp-calypso/assets/5634774/c5fa315a-c48f-46e6-a13a-c6dd46aef53b)

3) looks like this icon needs some love on hover (it should not be solid black):

![CleanShot 2024-03-22 at 09 21 03](https://github.com/Automattic/wp-calypso/assets/5634774/53c593c3-e326-4f36-9b36-a3a06f806574)

4) This text likely needs to be a middle gray (Maybe 40 or 50?):

![CleanShot 2024-03-22 at 09 22 32@2x](https://github.com/Automattic/wp-calypso/assets/5634774/b77ad94a-e032-43e8-aaaa-a735f353b52a)

5) Seeing a double outline on the notifications bell:

![CleanShot 2024-03-22 at 09 23 46@2x](https://github.com/Automattic/wp-calypso/assets/5634774/038313ca-6bba-480a-b26b-9bee1ab161b1)

6) Not a blocker, but @lucasmendes-design think you might be able to find an outlined version of this icon (for consistency with the other outlined icons)?

![CleanShot 2024-03-22 at 09 25 03@2x](https://github.com/Automattic/wp-calypso/assets/5634774/491c0909-4567-440c-a9ba-1e1287b7e7e7)

Addison-Stavlo committed:
stop a8c logo from black on hover

Addison-Stavlo committed:
fix icon hover colors on me

Addison-Stavlo committed:
update hover background between gray 0 and 5

Addison-Stavlo committed:
update p2 last updated items to use alternative text color, and updated alt text color

Addison-Stavlo committed:
update me sidebar gravatar user handle to use alternative text color

Addison-Stavlo committed:
fix alignment of create new list item

Addison-Stavlo commented:
Thanks @davemart-in !  I applied some changes this should be ready for another look.

- [ ]  Feels like we may want a new color in-between 0 and 5. In my prototype, I had --studio-gray-2-5: #eee; which feels like a good compromise.  - updated to #eee for the hover color
- [ ]  This text likely needs to be a middle gray (Maybe 40 or 50?): - Looking good at 40 now.
- [ ]  These my profile icons should probably be dark, not white on hover: - fixed, they were using the wrong variable and didnt take my recent change from white to dark text on hover.

> looks like this icon needs some love on hover (it should not be solid black):

Yeah, thats one of the 'minor issues' Im still looking at. It seems to have bad stroke colors in various states (all black sometimes, too dark gray others, bright white on others 🤪 .  Similarly, the "manage subscriptions" icon at the bottom is a bit inconsitent... sometimes its black when selected, sometimes not, sometimes the hover styles are causing it to be black when selected and hovered..

> Seeing a double outline on the notifications bell:

This I cant seem to reproduce. I saw it once when I initially made a change... but it went back to normal and I havent seen it since. Will investigate.

Addison-Stavlo committed:
fix p2 count being white on hover

davemart-in reviewed (approved):
I feel good about this. I'd feel okay just shipping it and we can adjust anything else that pops up in a separate PR.

Addison-Stavlo committed:
fix notification bell issue with unseen dot

Addison-Stavlo committed:
finalize styles

Addison-Stavlo committed:
fix issue with sites svg getting smaller after visitng me

Addison-Stavlo committed:
fix a8c icon going black issue again...

Addison-Stavlo committed:
fix issue with focus outlines on menu headings

Addison-Stavlo committed:
redefine dependent colors to use contextual vars

Addison-Stavlo committed:
get rid of blank line

Addison-Stavlo committed:
fix issue with selected menu groups not showing selected colors when closed
